cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/util/ProgramRunMetaFileUtil.java

```diff
@@ -0,0 +1,126 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.report.util;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.report.StartInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Utility class for reading and writing program run meta files.
+ */
+public class ProgramRunMetaFileUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(ProgramRunMetaFileUtil.class);
+
+  // TODO: [CDAP-13215] add omitted fields when the actual program meta files are generated
+  private static final Schema STARTING_INFO = Schema.recordOf(
+    "ProgramStartingInfo",
+    Schema.Field.of(Constants.USER, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.RUNTIME_ARGUMENTS, Schema.mapOf(Schema.of(Schema.Type.STRING),
+                                                              Schema.of(Schema.Type.STRING)))
+    );
+
+  // TODO: [CDAP-13215] add omitted fields when the actual program meta files are generated
+  private static final String SCHEMA_STRING = Schema.recordOf(
+    "ReportRecord",
+    Schema.Field.of(Constants.NAMESPACE, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.PROGRAM, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.RUN, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.STATUS, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.TIME, Schema.of(Schema.Type.LONG)),
+    Schema.Field.of(Constants.START_INFO,  Schema.nullableOf(STARTING_INFO))
+  ).toString();
+
+  public static final org.apache.avro.Schema STARTING_INFO_SCHEMA =
+    new org.apache.avro.Schema.Parser().parse(STARTING_INFO.toString());
+  public static final org.apache.avro.Schema SCHEMA = new org.apache.avro.Schema.Parser().parse(SCHEMA_STRING);
+
+  /**
+   * Adds mock program run meta files to the given location.
+   * TODO: [CDAP-13216] this method should be marked as @VisibleForTesting. Temporarily calling this method
+   * when initializing report generation Spark program to add mock data
+   *
+   * @param metaBaseLocation the location to add files
+   */
+  public static void populateMetaFiles(Location metaBaseLocation) throws Exception {
```

Comment:
Seems like this method is only for unit-testing? If that's the case, it should goes into `src/test/java`, not main.